packages/react-slider/src/components/Slider/useSlider.ts

```diff
@@ -1,36 +1,36 @@
 import * as React from 'react';
-import { makeMergeProps, resolveShorthandProps } from '@fluentui/react-utilities';
-import { SliderProps, SliderShorthandProps } from './Slider.types';
+import { resolveShorthand } from '@fluentui/react-utilities';
+import { SliderProps, SliderSlots, SliderState, SliderPublicRef } from './Slider.types';
+import { useSliderState } from './useSliderState';
 
 /**
  * Array of all shorthand properties listed in SliderShorthandProps
  */
-export const sliderShorthandProps: SliderShorthandProps[] = [
-  /* TODO add shorthand property names */
+export const sliderShorthandProps: Array<keyof SliderSlots> = [
+  'rail',
+  'trackWrapper',
+  'track',
+  'thumbWrapper',
+  'thumb',
+  'activeRail',
 ];
 
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-const mergeProps = makeMergeProps<any>({ deepMerge: sliderShorthandProps });
-
 /**
- * Create the state required to render Slider.
- *
- * The returned state can be modified with hooks such as useSliderStyles,
- * before being passed to renderSlider.
- *
- * @param props - props from this instance of Slider
- * @param ref - reference to root HTMLElement of Slider
- * @param defaultProps - (optional) default prop values provided by the implementing type
+ * Given user props, returns state and render function for a Slider.
  */
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-export const useSlider = (props: SliderProps, ref: React.Ref<HTMLElement>, defaultProps?: SliderProps): any => {
-  const state = mergeProps(
-    {
-      ref,
-    },
-    defaultProps && resolveShorthandProps(defaultProps, sliderShorthandProps),
-    resolveShorthandProps(props, sliderShorthandProps),
-  );
+export const useSlider = (props: SliderProps, ref: React.RefObject<HTMLElement & SliderPublicRef>): SliderState => {
```

Comment:
I don't think the typings for `ref` are correct--unless I'm missing something, it's not possible for the ref to simultaneously point to a native element and an imperative handle.